src/fields/field_set.cc

```diff
@@ -30,23 +30,25 @@ FieldSet::FieldSet()
 : mesh_(nullptr) {}
 
 
+const Input::Type::Selection & FieldSet::get_user_field_shape_selection() {
+	return Input::Type::Selection("User_fields_shape", "Allowed shapes of user fields.")
+		.add_value(FieldSet::scalar, "scalar", "Scalar user field.")
+		.add_value(FieldSet::vector, "vector", "Vector user field.")
+		.add_value(FieldSet::tensor, "tensor", "Tensor user field.")
+		.close();
+}
+
 const Input::Type::Record & FieldSet::make_user_field_type(const std::string &equation_name) {
     static Field<3, FieldValue<3>::Scalar> scalar_field;
     static Field<3, FieldValue<3>::VectorFixed> vector_field;
     static Field<3, FieldValue<3>::TensorFixed> tensor_field;
     return Input::Type::Record( equation_name+":UserData", "Record to set fields of the equation: "+equation_name+".")
         .declare_key("name", Input::Type::String(), Input::Type::Default::obligatory(),
                      "Name of user defined field.")
-        .declare_key("is_boundary", Input::Type::Bool(), Input::Type::Default("false"),
-                     "Type of field: boundary or bulk.")
-        .declare_key("scalar_field", scalar_field.get_input_type(),
-                     "Instance of FieldAlgoBase ScalarField descendant.\n"
-        		     "One of keys 'scalar_field', 'vector_field', 'tensor_field' must be set.\n"
-        		     "If you set more than one of these keys, only first key is accepted.")
-        .declare_key("vector_field", vector_field.get_input_type(),
-                     "Instance of FieldAlgoBase VectorField descendant. See above for details.")
-        .declare_key("tensor_field", tensor_field.get_input_type(),
-                     "Instance of FieldAlgoBase TensorField descendant. See above for details.")
+        .declare_key("shape_type", FieldSet::get_user_field_shape_selection(), Input::Type::Default("\"scalar\""), "Shape of user field.")
```

Comment:
Add bparser error test, for the case the user does not specify the shape for a tensor valued field formula user field. I guess that an error in bparser would be reported. We have to test the quality of bparser errors in the flow123d context.